packages/ingest/test/test-HttpsProviderClient.js

```diff
@@ -67,17 +102,72 @@ test('download() downloads a file', async (t) => {
 test('sync() downloads remote file to s3 with correct content-type', async (t) => {
   const bucket = randomString();
   const key = 'syncedFile.json';
-  const remotePath = 'test/file.hdf';
-  const remoteContent = '<HDF CONTENT>';
-  const expectedContentType = 'application/x-hdf';
-  t.context.server.get(`/${remotePath}`, (_, res) => {
-    res.header({ 'content-type': expectedContentType });
-    res.end(remoteContent);
-  });
+
   try {
     await s3().createBucket({ Bucket: bucket }).promise();
     await t.context.httpsProviderClient.sync(
-      remotePath, bucket, key
+      publicFile, bucket, key
+    );
+    t.truthy(fileExists(bucket, key));
+    const syncedContent = await getTextObject(bucket, key);
+    t.is(syncedContent, remoteContent);
+
+    const s3HeadResponse = await headObject(bucket, key);
+    t.is(expectedContentType, s3HeadResponse.ContentType);
+  } finally {
+    await recursivelyDeleteS3Bucket(bucket);
+  }
+});
+
+test('HttpsProviderClient throws error if it gets a username but no password', (t) => {
+  t.throws(() => new HttpProviderClient({
+    protocol: 'https',
+    host: '127.0.0.1',
+    port: t.context.server.sslPort,
+    certificateUri: `s3://${t.context.configBucket}/certificate.pem`,
+    username: 'user'
+  }),
+  {
+    instanceOf: ReferenceError,
+    message: 'Found providerConfig.username, but providerConfig.password is not defined'
+  });
+});
+
+test('HttpsProviderClient supports basic auth with redirects for download', async (t) => {
+  const httpsProviderClient = new HttpProviderClient({
+    protocol: 'https',
+    host: '127.0.0.1',
+    port: t.context.server.sslPort,
+    certificateUri: `s3://${t.context.configBucket}/certificate.pem`,
+    username: basicUsername,
+    password: basicPassword
+  });
+
+  const localPath = path.join(tmpdir(), randomString());
+  try {
+    await httpsProviderClient.download(protectedFile, localPath);
+    t.is(fs.existsSync(localPath), true);
+  } finally {
+    fs.unlinkSync(localPath);
+  }
+});
+
+test('HttpsProviderClient supports basic auth with redirects for sync', async (t) => {
```

Comment:
Is there a test in here for using encrypted credentials with basic auth?

Reply:
I have yet to figure out how to adequately test this. Should I mock out `decrypt`, or go the end-to-end route with full KMS testing?

Reply:
I would mock out `decrypt`. Proving that the decryption works is a separate unit test. So the test here is "end to end" in the sense that it proves that encrypted credentials are unencrypted before being used for auth and a separate test verifies that the decryption itself works.